### DISCUSSION TITLE
how to use github as a repository

### DISCUSSION THREAD

**Discussion by gv29847** [community/community#82718] [New to GitHub]
I wrote an add on for Liberty Bassic a year or two back and would like to make it freely available to anyone interested.I don't want to use my own web site since at 83 a am less sure of my ability to keep the site going than most of you. 
Can I use github for this purpose, and if so how do I go about it?
How do I access any replies?

**Comment by rishiskhare** [comment]
Yes, you can! I would suggest uploading your project to GitHub as a repository if you'd like to make it available to the public. I suggest using the following video: https://www.youtube.com/watch?v=vpRkAoCqX3o

**Reply by gv29847** [reply]
Thanks for this - the video was transparent right up to the end. The last shot shows the files in github in a panel called "codebase". Howevever the window I see is the same as shown in the video - except the "codebase" panel is replaced by "Set up GitHub Copilot". So how do I (or anyone) see what is in the repository? Also to add files can I just copy / paste between win and my GH repository? And delete?

**Reply by gv29847** [reply]
Sorry, the previous was misleading - I had not looked at what was in the Command prompt window as I should have.
At the ">git add ." I got 
>git commit -m "first commit"
Author identity unknown
*** Please tell me who you are.
Run
  git config --global user.email "you@example.com"
  git config --global user.name "Your Name"

And at 
LBPP for Github>git push -u origin main
error: src refspec main does not match any
error: failed to push some refs to 'https://github.com/gv29847/LBPrePass.git'

**Comment by anuragdaksh7** [comment]
1. Install git bash
2. open the terminal in project directory
3. git init
4. git add -A
5. git commit -m "commit message"
6. git remote add origin "your repository url"
7. git push

**Reply by gv29847** [reply]
Thanks for the reply - but youe are above me - I do not understand steps 3- 7

**Reply by anuragdaksh7** [reply]
on your terminal (opened in the project directory) execute these commands

**Comment by varunpats** [comment]
Hey @gv29847,

After opening project directory in terminal. Type following commands step by step, 

- git init (This commands creates an empty Git repository or reinitialize an existing one) 
- git add -A (This will add file contents to the index )
- git commit -m "commit message" (This command used to commit changes with the message of your choice in double quotes after "-m")
- git push (To push changes to your github repo)

Hope this helps! If you have any doubts please continue in comments.

**Comment by gv29847** [comment]
Thanks for the explanations.
Did all that, but had to add
C:\Users\emad\Documents\Liberty PrePass\LBPP for Github>git config --global user.email [29847gv@gmail.com](mailto:29847gv@gmail.com)
C:\Users\emad\Documents\Liberty PrePass\LBPP for Github>git config --global user.name "petermat"
After git commit
And got 
C:\Users\emad\Documents\Liberty PrePass\LBPP for Github>git push
fatal: The current branch main has no upstream branch.
To push the current branch and set the remote as upstream, use
    git push --set-upstream origin main
To have this happen automatically for branches without a tracking
upstream, see 'push.autoSetupRemote' in 'git help config'.
Is this relevant / important?

Also when logging into github.com I cannot find a way of seeing what is in the repository

**Comment by gv29847** [comment]
Varun,

Thanks for the explanations.

Did all that, but had to add

C:\Users\emad\Documents\Liberty PrePass\LBPP for Github>git config --global user.email ***@***.*** ***@***.***> 

C:\Users\emad\Documents\Liberty PrePass\LBPP for Github>git config --global user.name "petermat"

After git commit

And got 

C:\Users\emad\Documents\Liberty PrePass\LBPP for Github>git push

fatal: The current branch main has no upstream branch.

To push the current branch and set the remote as upstream, use

    git push --set-upstream origin main

To have this happen automatically for branches without a tracking

upstream, see 'push.autoSetupRemote' in 'git help config'.

Is this relevant / important?

 

Also when logging into github.com I cannot find a way of seeing what is in the repository

 

Peter

**Comment by gv29847** [comment]
OK I found my repository - you have to say you want copilot and then say you dont! Also I have now dragged and dropped sucessfully so I think I am good to go for now. Thanks everyone for the help.
